posts/2023-10-17-23.0.0.10.adoc

```diff
@@ -0,0 +1,162 @@
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/26423
+// Contact/Reviewer: gjwatts
+// // // // // // // // 
+[#java21]
+== Support for Java 21
+// The following excerpt for issue https://github.com/OpenLiberty/open-liberty/issues/26192 was found in 2023-09-26-23.0.0.10-beta.adoc.
+// ------ <Excerpt From Previous Post: Start> ------
+// Contact/Reviewer: gjwatts
+// // // // // // // // 
+
+Java 21 is finally here, the first long-term support (LTS) release since Java 17 was released two years ago. It offers some new functionality and changes that you'll want to check out for yourself.
+
+Take advantage of trying out the new changes in Java 21 now and to test out your applications, microservices, and runtime environments.
+
+To run Open Liberty with Java 21:
+
+1. Download and install link:https://openliberty.io/start/#runtime_releases[Open Liberty 23.0.0.10], or later.
+2. Download the latest release of Java 21 from link:https://adoptium.net/temurin/releases/?version=21[adoptium.net].
+3. Edit your Open Liberty runtime link:/docs/latest/reference/config/server-configuration-overview.html#server-env[server.env file] to point `JAVA_HOME` to your Java 21 installation.
+4. Start testing!
+
+Here are some highlights from new JEP changes in Java 18-21:
+
+* 400: link:https://openjdk.java.net/jeps/400[UTF-8 by Default]
+* 408: link:https://openjdk.java.net/jeps/408[Simple Web Server]
+* 413: link:https://openjdk.java.net/jeps/413[Code Snippets in Java API Documentation]
+* 416: link:https://openjdk.java.net/jeps/416[Reimplement Core Reflection with Method Handles]
+* 418: link:https://openjdk.java.net/jeps/418[Internet-Address Resolution SPI]
+* 421: link:https://openjdk.java.net/jeps/421[Deprecate Finalization for Removal]
+* 422: link:https://openjdk.org/jeps/422[Linux/RISC-V Port]
+* 431: link:https://openjdk.org/jeps/431[Sequenced Collections]
+* 439: link:https://openjdk.org/jeps/439[Generational ZGC]
+* 440: link:https://openjdk.org/jeps/440[Record Patterns]
+* 441: link:https://openjdk.org/jeps/441[Pattern Matching for switch]
+* 449: link:https://openjdk.org/jeps/449[Deprecate the Windows 32-bit x86 Port for Removal]
+* 451: link:https://openjdk.org/jeps/451[Prepare to Disallow the Dynamic Loading of Agents]
+* 452: link:https://openjdk.org/jeps/452[Key Encapsulation Mechanism API]
+
+But perhaps the most anticipated one of all is the introduction of Virtual Threads in Java 21:
+
+* 444: link:https://openjdk.org/jeps/444[Virtual Threads]
+
+Will the impact of Virtual Threads live up to the anticipation?  Find out for yourself by trying them out in your applications run on the best Java runtime, Open Liberty!
```

Comment:
should we provide a sentence or two here to summarize the value/impact hype around virtual threads? Can reduce memory consumption, improve app availability and throughput etc-

Reply:
Virtual threads were designed to provide higher throughput for running tasks that spend most of their time blocked, like waiting for I/O operations.